protocol-units/da/m1/light-node/src/v1/sequencer.rs

```diff
@@ -75,23 +80,79 @@ impl LightNodeV1 {
 			}
 			None => {
 				// no transactions to include
-				info!(target: "movement_timing", uid = %uid, "no_transactions_to_include");
+				debug!(target: "movement_timing", uid = %uid, "no_transactions_to_include");
 				Ok(())
 			}
 			
 		}
 
 	}
 
-	async fn tick_publish_blobs(&self, receiver : &mut Receiver<Block>) -> Result<(), anyhow::Error> {
-		
-		let half_building_time = self.memseq.building_time_ms();
-		let start = std::time::Instant::now();
-		let mut blocks = Vec::new();
+	/// Attempts to submit a batch of blobs according to the chunk size
+	async fn submit_and_resize(&self, block_blobs: &mut Vec<celestia_types::Blob>, chunk_size : usize) -> Result<usize, anyhow::Error> {
+	
+
+		// iter in chunks of 4
+		while !block_blobs.is_empty() {
+			let mut chunk: Vec<celestia_types::Blob> = Vec::with_capacity(chunk_size);
+			
+			for _ in 0..chunk_size {
+				if let Some(blob) = block_blobs.pop() {
+					chunk.push(blob);
+				} else {
+					break;
+				}
+			}
```

Comment:
Could be more concise and preserve the order of blobs:
```suggestion
			let chunk = block_blobs.drain(..min(chunk_size, block_blobs.len())).collect::<Vec<_>>();
```

If the order is not important, `drain` can take from the end, it will work a bit faster than popping one by one.